src/app/core/export/export-config.controller.js

```diff
@@ -44,104 +43,165 @@ exports.requestExport = (req, res) => {
+				});
+				stream.push(null);
+			})
+		);
+	}
+};
+
+/**
+ * Build a readable stream from data and pipe through a chain of transform streams
+ *
+ * @param {streams.Readable | *} data the data to be exported, either in the form of a readable stream or an object accompanied by a getRead function
+ * @param {Function} getRead a function that takes a readable stream and returns the _read function for the new stream if data is not a readable stream
+ * @param {streams.Transform[]} [transforms] an optional array of transform streams through which to pipe the export data
+ */
+const buildExportStream = (data, getRead, transforms) => {
+	let stream = data;
+
+	if (!(stream instanceof streams.Readable)) {
+		stream = new streams.Readable({ objectMode: true });
+		stream._read = getRead(stream);
+	}
+
+	if (!stream.destroy) {
+		stream.destroy = () => {
+			stream.destroyed = true;
 		};
+	}
 
-		this.exportCSVFromStream(req, res, filename, columns, s);
+	if (transforms && transforms.length) {
+		// reduce the initial stream and transform streams to a single stream
+		// destroying the resulting stream will also destroy all of the transform streams and the initial streams
+		stream = transforms.reduce((prevStream, transform) => {
+			// pipe previous stream through current transform stream
+			const newStream = prevStream.pipe(transform);
+
+			// if the previous stream has a defined `destroy()` function, we need to combine it with the newStream's destroy function
+			if (prevStream.destroy) {
+				// save the potentially undefined `destroy()` function from the new stream
+				const originalDestroy = newStream.destroy;
+
+				// we need to destroy both the new stream and the previous stream here
+				newStream.destroy = () => {
+					// newStream had a `destroy()` function already defined, use it
+					if (originalDestroy) {
+						originalDestroy.apply(this);
+					} else {
+						// the new stream did not have a defined `destroy()` function,
+						//  so we should mark it as `destroyed` at this point
+						newStream.destroyed = true;
+					}
+
+					// now destroy the previous stream
+					prevStream.destroy();
+				};
+			}
+
+			return newStream;
+		}, stream);
 	}
+
+	return stream;
 };
 
 /**
- * Export a CSV file with rows derived from a readable stream
+ * Export to a file from a readable stream
  *
- * @param {Request} req
- * @param {Response} res
- * @param {string} filename
- * @param {*} columns
- * @param {ReadableStream} stream a readable stream containing data for rows
+ * @param {*} req
+ * @param {*} res
+ * @param {string} fileName
+ * @param {'csv' | 'plain'} fileType
+ * @param {streams.Readable} stream
  */
-exports.exportCSVFromStream = (req, res, filename, columns, stream) => {
-	res.set('Content-Type', 'text/csv;charset=utf-8');
-	res.set('Content-Disposition', `attachment;filename="${filename}"`);
+const exportStream = (req, res, fileName, fileType, stream) => {
+	res.set('Content-Type', `text/${fileType};charset=utf-8`);
```

Comment:
We probably want to just require that the use of this to provide the full MIME-type header rather than 'just' the `filetype`.
I could imagine a case down the road where we want to export with a content type of something like `application/json`.